Add readme description (#1)
sergepilipchuk/maui-chat-control: README.md

<!-- default badges list -->
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # DevExpress Chat for .NET MAUI

Reviewer: This title looks like we supply a Chat component that's specifically designed for building chats. But it's just an example of how you can use our basic components and templates to build such a view. I would make that clear.

```suggestion
# Use DevExpress .NET MAUI Components to Build a Chat View
```

> This example demonstrates a Chart control that allows users to send and receive massages.

Reviewer: I wouldn't say here that the control "allows to send/receive message". The example, as I see, only focuses on building the UI shell. The messages are never sent or received - they are just a collection in the model. No data is transferred between different clients. 

```suggestion
This example uses DevExpress .NET MAUI Components to display a chat view with sender and receiver messages, suggested quick replies, and response input controls.
```

* [SafeKeyboardAreaView](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.SafeKeyboardAreaView)
* [ChipGroup](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChipGroup): [ChipTapCommand](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChipGroup.ChipTapCommand), [ItemsSource](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChipGroup.ItemsSource), [IsMultiline](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChipGroup.IsMultiline), [DisplayMember](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.ChipGroup.DisplayMember)
* [TextEdit](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEdit): [PlaceholderText](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.PlaceholderText), [Text](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEditBase.Text)
* [DXButton](https://docs.devexpress.devx/MAUI/DevExpress.Maui.Core.DXButton): [Command](https://docs.devexpress.com/MAUI/DevExpress.Maui.Core.DXButtonBase.Command), [Icon](https://docs.devexpress.devx/MAUI/DevExpress.Maui.Core.DXContentPresenter.Icon)

## Implementation Details

> 

Reviewer: I would expand the description a little bit, if only to include all controls that you listed in the section above. It makes sense from that point of view, and also it seems a bit strange that we never talk about individual items or data sources in Implementation Details. 

So let's start with `SafeKeyboardAreaView`. 

```suggestion
1. Use a `SafeKeyboardAreaView` as the root for the view layout. This way, the device keyboard will not overlap the message view when it appears.

    ```xaml
    <dx:SafeKeyboardAreaView> 
        ... 
    </dx:SafeKeyboardAreaView>
    ```

> 1. Use the `DXCollectionView` control to display messages and group them by **DateRange**:

Reviewer: ```suggestion
1. Use the `DXCollectionView` control to display messages. Specify the data source and item templates (use different templates for sender and receiver). Groups items using a built-in algorithm that uses date/time ranges ("Today", "Yesterday", "Last Week", and so on). Specify a template for group headers. 
```

    ```xaml

>     <dxcv:DXCollectionView ... 
>         x:Name="chatSurface">
>         <dxcv:DXCollectionView.GroupDescription>
>             <dxcv:GroupDescription ... 
>                 GroupInterval="DateRange" />
>         </dxcv:DXCollectionView.GroupDescription>
>     </dxcv:DXCollectionView>

Reviewer: ```suggestion
        <dxcv:DXCollectionView
        ItemsSource="{Binding Messages}"
        ItemTemplate="{local:MessageTemplateSelector SenderTemplate=..., RecipientTemplate=...}"
        GroupHeaderTemplate="{StaticResource dayGroupTemplate}"
        ...
        >
            <dxcv:DXCollectionView.GroupDescription>
                <dxcv:GroupDescription FieldName="SentAt" GroupInterval="DateRange" />
            </dxcv:DXCollectionView.GroupDescription>
        </dxcv:DXCollectionView>
```

    ```

> 

Reviewer: Let's add a step that mentions `DXContentPresenter` and message templates: 

```suggestion
2. Use `DXContentPresenter` components to define templates for sender and receiver messages. 

    ```xaml
        <ContentPage.Resources>
                <DataTemplate x:Key="senderMessageTemplate"  x:DataType="{x:Type local:Message}">
                    <dx:DXContentPresenter ... >
                    </dx:DXContentPresenter>
                </DataTemplate>
                <DataTemplate x:Key="recipientMessageTemplate" x:DataType="{x:Type local:Message}">
                    <dx:DXContentPresenter ... >
                    </dx:DXContentPresenter>
                </DataTemplate>
                ...
        </ContentPage.Resources>
    ```
```

        // ...
        void OnMessagesCollectionChanged(object sender, NotifyCollectionChangedEventArgs e) {
            chatSurface.ScrollTo(chatSurface.GetItemHandle(vm.Messages.Count - 1));
        }
    }
    ```
    
3. Use the [`ChipGroup`](https://docs.devexpress.devx/MAUI/DevExpress.Maui.Editors.ChipGroup) control to display short answers:

    ```xaml
    <dxe:ChipGroup ...      
        DisplayMember="Text"
        ItemsSource="{Binding SuggestedActions}" />
    ```

    ```csharp
    public ChatViewModel() {
        // ...
        SuggestedActions = new ObservableCollection<SuggestedAction>() {
            new SuggestedAction() { Message = new Message() { Sender = Me, SentAt = DateTime.Now, Text = "Sure" }, Text = "Sure" },
            new SuggestedAction() { Message = new Message() { Sender = Me, SentAt = DateTime.Now, Text = "Great" }, Text = "Great" },
            new SuggestedAction() { Message = new Message() { Sender = Me, SentAt = DateTime.Now, Text = "Thank you" }, Text = "Thank you" },
            new SuggestedAction() { Message = new Message() { Sender = Me, SentAt = DateTime.Now, Text = "My pleasure" }, Text = "My pleasure" }
        };
    }

>     ```

Reviewer: And let's add a step that mentions Input and Button - controls that are supposed to send a new message.
